models/classes/CrudResultsService.php

```diff
@@ -181,4 +188,23 @@ protected function getClassService()
     {
         // TODO: Implement getClassService() method.
     }
+
+    /**
+     * @param $variable
+     * @return array
+     * @throws common_exception_Error
+     */
+    protected function formResource($variable){
+        $resource['value'] = $variable->getValue();
+        $resource['identifier'] = $variable->getIdentifier();
+        if ($variable instanceof taoResultServer_models_classes_ResponseVariable) {
+            $resource['type'] = new core_kernel_classes_Class('http://www.tao.lu/Ontologies/TAOResult.rdf#ResponseVariable');
+        } else {
+            $resource['type']= new core_kernel_classes_Class('http://www.tao.lu/Ontologies/TAOResult.rdf#OutcomeVariable');
+        }
+        $resource['epoch'] = $variable->getEpoch();
+        $resource['cardinality'] = $variable->getCardinality();
+        $resource['basetype'] = $variable->getBaseType();
+        return $resource;
```

Comment:
does it make sense to use DTO instead of error-prone array approach?

Reply:
This part was moved from function to decrease mess. Array used on all upper levels.